tests/testthat/test-scale.r

```diff
@@ -19,3 +19,36 @@ test_that("NA.value works for discrete", {
   expect_that(dscale(x, pal, "grey50")[1], equals("grey50"))
   expect_that(dscale(x, pal, "grey50")[5], equals("grey50"))
 })
+
+test_that(paste(
+  "train_continuous maintains class with `existing=NULL`,",
+  "and changes class to the class of `existing` when not NULL."
+), {
+  expect_equal(
+    train_continuous(1:5),
+    c(1, 5)
+  )
+  expect_error(train_continuous(LETTERS[1:5]),
+    regexp = "Discrete value supplied to continuous scale"
```

Comment:
This regexp is a bit long - the point is to check that you're got the right error, not to validate the exact text of the error message.

Reply:
But why is it in this test? It doesn't seem to be related to the topic of test.

Reply:
Good point, I've separated it into another test (and while not technically related to this PR's main topic, it is related to increasing test coverage for `train_continuous()`).